openmrs/src/main/java/org/motechproject/openmrs/domain/Visit.java

```diff
@@ -0,0 +1,145 @@
+package org.motechproject.openmrs.domain;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import java.lang.reflect.Type;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * Represents a single OpenMRS visit.
+ */
+public class Visit {
+
+    private String uuid;
+    private Location location;
+    private VisitType visitType;
+    private Date startDatetime;
+    private Date endDatetime;
+    private Patient patient;
+
+    /**
+     * Default constructor.
+     */
+    public Visit() {
+    }
+
+    public Visit(Date startDatetime, Date endDatetime, Patient patient, VisitType visitType) {
+        this.startDatetime = startDatetime;
+        this.endDatetime = endDatetime;
+        this.patient = patient;
+        this.visitType = visitType;
+    }
+
+    public Visit(Date startDatetime, Date endDatetime, Patient patient, VisitType visitType, Location location) {
```

Comment:
please set field values in this constructor and invoke it from the above one. It is more readable solution